distributed/client.py

```diff
@@ -602,9 +602,29 @@ def set_error(self, exception, traceback):
             The exception
         traceback: Exception
             The traceback
+        erred_on: set
+            Workers where it errored
+        key: tuple
+            Task key
         """
         _, exception, traceback = clean_exception(exception, traceback)
 
+        if (
+            dask.config.get("admin.traceback.shorten.when", None)
+            and dask.config.get("admin.traceback.shorten.what", None)
+            and key
+            and erred_on
```

Comment:
```suggestion
            key
            and erred_on
            and dask.config.get("admin.traceback.shorten.when")
            and dask.config.get("admin.traceback.shorten.what")
```

Reply:
I'm not disabling worker information. I'm disabling pushing worker information into exception's args.

As you correctly noted, what's going on below is hacky. I want this work with any `Exception`, and without overriding the built-in exception formatter, whether it's python console, ipython console, or ipython notebook. Whatever args were previously attached to the Exception, this will stringify the first arg, and append the worker info to it. This way, the worker info becomes an integral part of the exception, it doesn't have to be a custom exception, and it doesn't add another wrapping layer.